test-requirements-27.txt

```
@@ -44,6 +45,7 @@ Pygments==2.5.2
 pyparsing==2.4.7
 pyrsistent==0.15.7
 python-dateutil==2.8.1
+pywinpty==0.5.7  # some dependency (pywinpty>=0.5)
```

```suggestion
pywinpty==0.5.7  # any pywinpty>=0.5
```
Why did you add this?

---

To make the py2.7 tests work, if I don't add them pip will try to install packages that depend on the wheel (57 if I remember correctly) that are not supported by py2.7

---

For other OS, there were no such problems, because pywinpty is installed only when the OS is Windows.

---

By "some dependency" I mean that this package is installed during installation, but I do not know which other package depends on it and installs it

---

Speaking of 2.7, that's currently excluded from the build matrix.

---

The log output of pip will tell you why it included a specific package.

---

> currently excluded from the build matrix

Because I excluded it :) since I couldn't get it to work in a simple way.

---

> The log output of pip will tell you why it included a specific package

ok, judging by [this](https://github.com/0dminnimda/cython/runs/3214391018#step:5:163) output say it to me

I do not claim that this is exactly a dependency of some package, and therefore it is installed, but I did not find the source due to which this package is installed ...
Therefore, I assumed that this is a dependency of some package. ¯\_(ツ)_/¯

---

This is a log run *after* you added the package as a direct dependency. In that case, pip will tell you that it installs a direct dependency (surprise!). I meant the log output *before* you added the dependency. That would tell you why it installs the package as a transitive dependency.

---

It's a dependency of `terminado`, which is a dependency of Jupyter. Looks like we have to keep it then and just restrict the version, as you did.

---

I will change the comment accordingly

---

[resolved](https://github.com/cython/cython/pull/4324/commits/10938be9672759037ad534cd2d6bf0fd47e36c2d)